core/src/test/java/org/infinispan/stream/BaseStreamIteratorEvictionTest.java

```diff
@@ -0,0 +1,58 @@
+package org.infinispan.stream;
+
+import org.infinispan.Cache;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.filter.CollectionKeyFilter;
+import org.infinispan.filter.KeyFilterAsKeyValueFilter;
+import org.testng.annotations.Test;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+/**
+ * Base class to test if evicted entries are not returned via stream
+ *
+ * @author wburns
+ * @since 8.0
+ */
+@Test(groups = "functional", testName = "stream.BaseStreamIteratorEvictionTest")
+public abstract class BaseStreamIteratorEvictionTest extends BaseSetupStreamIteratorTest {
+   public BaseStreamIteratorEvictionTest(boolean tx, CacheMode mode) {
+      super(tx, mode);
+   }
+
+   public void testExpiredEntryNotReturned() throws InterruptedException {
+      Cache<Object, String> cache = cache(0, CACHE_NAME);
+      // First put some values in there
+      Map<Object, String> valuesInserted = new LinkedHashMap<Object, String>();
+      for (int i = 0; i < 5; ++i) {
+         Object key = i;
+         String value = key + " stay in cache";
+         cache.put(key, value);
+         valuesInserted.put(key, value);
+      }
+
+      int expectedTime = 2;
```

Comment:
Yeah actually this test should be tweaked to use TimeService instead.  I can add that as a TODO in the other issue.